st2common/st2common/service_setup.py

```diff
@@ -341,7 +342,10 @@ def register_service_in_service_registry(service, capabilities=None, start_heart
         'Joining service registry group "%s" as member_id "%s" with capabilities "%s"'
         % (group_id, member_id, capabilities)
     )
-    return coordinator.join_group(group_id, capabilities=capabilities).get()
+    try:
+        return coordinator.join_group(group_id, capabilities=capabilities).get()
+    except MemberAlreadyExist:
+        pass
```

Comment:
More comment on legacy, but in the try we return whatever coordinator.join_group returns, but if memberalreadyexists we don't return anything...
I think in fact the join_group doesn't return anything - so probably just worth changing it so that we change the try to remove the "return" on the call to coordinator.join_group. Or if its expected to return something then we need to adjust the except clause...